subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/MutableModuleComponentResolveMetadata.java

```diff
@@ -84,6 +84,10 @@
 
     void setAttributes(AttributeContainer attributes);
 
+    boolean isRequiresAttributeMatching();
```

Comment:
:x: I don't like the name much. Should be `requiresAttributeMatching()` or `isAttributeMatchingRequired()`.

Also can you add javadocs? We need to make sure we know why this is needed.

Reply:
How about `attributeMatchingRequested`? https://github.com/gradle/gradle/pull/4016/commits/4a400338a9e79ae7307c65b8b785ad7f1eb03f18